lib/learn_to_rank/ctr_to_judgements.rb

```diff
@@ -29,15 +29,34 @@ def relevancy_judgements
     attr_reader :ctrs
 
     def ctr_to_relevancy_score(query_ctrs)
+      benchmark = 0.0
       query_ctrs.each_with_object({}) do |(position, ctr), h|
         pos = Float(position)
         ctr = Float(ctr)
+
+        # Generate a normalised score based on CTR
         k = (2 * (Math.log(pos + 1) + 1) * ctr ** 0.5) / 5
-        h[position] = if k < 0.1
+
+        if k > 3
+          # If value is above three we likely have an edge case with high rates
+          h[position] = 3
+          next
+        elsif k > benchmark
+          # Sets a new "high point" for searches and places it at top
+          benchmark = k
+          h[position] = 3
+          next
+        end
+
+        # Calculates relevancy score based on percentage of relevance to the
+        # benchmark. Allows fine tuning of boundaries and takes into account
+        # weak searches where highest judgment is k < 2. Risks flattening the
+        # results if CTR calculation results are close.
+        h[position] = if k < (benchmark / 100) * 30
           0
-        elsif k < 1
+        elsif k < (benchmark / 100) * 60
           1
-        elsif k < 2
+        elsif k < (benchmark / 100) * 90
```

Comment:
I think you'd need `100.0` to be a float. Otherwise you might lose some precision, as `(4 / 100.0) * 60 = 2.4` and `(4 / 100) * 60 = 0`

Reply:
Yeah I misremembered how ruby handles types. Fixed shortly.